src/openforms/submissions/models.py

```diff
@@ -528,6 +532,29 @@ def payment_registered(self) -> bool:
         # TODO support partial payments
         return self.payments.filter(status=PaymentStatus.registered).exists()
 
+    def get_appointment_step_url(self) -> str:
+        for form_step in self.form.formstep_set.all():
+            for component in form_step.iter_components(recursive=True):
+                if component.get("appointments", {}).get("showProducts"):
+                    return form_step.form_definition.slug
+
+        return ""
```

Comment:
This is appointment specific logic/functionality that belongs in the `appointments` app. Should be a utility/service function from that app that's called where you need it. Do not overload different apps with functionality from other apps and keep them as decoupled as possible.